### DISCUSSION TITLE
SpriteBatch

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [amerkoleci/Vortice.Windows#346] [Q&A] [answered]
I don't see  spritebatch code in Vortice source code on Github, as seen here...

(https://docs.microsoft.com/en-us/windows/win32/api/d2d1_3/nn-d2d1_3-id2d1spritebatch)

But I do see Vortice.Direct2D1.ID2D1SpriteBatch in VS.

What is the situation here? I'm trying to work out how to use it, and I was hoping the source code would guide me. I found some sharpdx code, but sharpdx is a bit different. And I don't understand why I can see the types in my project but not in the Vortice.Windows repository.

![SpriteBatchSearch](https://user-images.githubusercontent.com/4374176/185101559-f622d7b3-3bee-4836-9aa4-c0030ec88a55.png)

**Comment by amerkoleci** [comment, marked as answer]
You don't have to look in repo but in assembly, there is https://www.fuget.org/packages/Vortice.Direct2D1/2.1.19/lib/netstandard2.0/Vortice.Direct2D1.dll/Vortice.Direct2D1/ID2D1SpriteBatch, you can use it for sure
